device-registry/src/main/scala/org/genivi/sota/device_registry/GroupsResource.scala

```diff
@@ -69,7 +78,21 @@ class GroupsResource(namespaceExtractor: Directive1[Namespace])
     complete(db.run(GroupInfoRepository.renameGroup(groupId, newGroupName)))
 
   def countDevices(groupId: Uuid): Route =
-    complete(db.run(GroupMember.countDevicesInGroup(groupId)))
+    complete(db.run(GroupMemberRepository.countDevicesInGroup(groupId)))
+
+  def addDeviceToGroup(groupId: Uuid, namespace: Namespace, deviceId: Uuid): Route = {
+    val dbIO = for {
+      _ <- DeviceRepository.exists(namespace, deviceId)
+      _ <- GroupInfoRepository.exists(groupId, namespace)
+    } yield ()
```

Comment:
Checking the group belongs to the namespace twice, better to do this check only when we extract info.